### PR TITLE
Feat/on 77 populate case details

### PR DESCRIPTION
**Pull request by tongzxc**

### CODE REVIEW

File: src/components/CaseDetails.vue [hunk 1]
```diff
@@ -0,0 +1,368 @@
+<template>
```
Comment (farablanco) [inline src/components/CaseDetails.vue]: what is this file for ?
remove the file if this is not related to  the case details page.

File: src/pages/Cases.vue [hunk 2]
```diff
@@ -171,8 +171,8 @@ ${state.url}`)
 
       // For each data, transform it to the desired shape
       const transformedData = fetchedData.map((data) => {
-        console.log(`typeof:referee`, typeof data.referee)
-        console.log('is it null?', data.referee === null)
+        // console.log(`typeof:referee`, typeof data.referee)
```
Comment (farablanco) [inline src/pages/Cases.vue]: just delete this instead of commenting

File: src/pages/Details.vue [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
   <div class="parent">
+
+    <!-- <script type="text/javascript">
+    const CaseNumber = 
+    </script> -->
     <div class="left">
       <div class="top">
-        <div class="title">Case EF 2020-0104</div>
+        <div class="title">Case #{{caseDetails.caseNumber}}</div>
+        <div v-bind="test" />
```
Comment (farablanco) [inline src/pages/Details.vue]: remove

File: src/pages/Details.vue [hunk 4]
```diff
@@ -37,55 +42,67 @@
...
-                <ul>
-                  Email
-                </ul>
+              <div class="sectionBody">
+                <div class="contentLeft">
+                  <ul>
+                    Name
+                  </ul>
+                  <ul>
+                    Occupation
```
Comment (farablanco) [inline src/pages/Details.vue]: Organization

File: src/pages/Details.vue [hunk 5]
```diff
@@ -37,55 +42,67 @@
...
                 </ul>
                 <ul>
-                  21/08/2021
+                  {{caseDetails.lastUpdated}}
                 </ul>
               </div>
             </div>
           </div>
           <div class="border">
             <div class="sectionHeading">REFEREE</div>
```
Comment (farablanco) [inline src/pages/Details.vue]: REFERENCE

File: src/pages/Details.vue [hunk 6]
```diff
@@ -37,55 +42,67 @@
...
+                  <ul>
+                    {{caseDetails.refereeEmail}}
+                  </ul>
+                </div>
               </div>
-              <div class="contentRight"></div>
-            </div>
           </div>
           <div class="border">
             <div class="sectionHeading">Referee Status</div>
```
Comment (farablanco) [inline src/pages/Details.vue]: Reference Status

File: src/pages/Details.vue [hunk 7]
```diff
@@ -155,7 +188,29 @@
...
+          <ul>
+            {{caseDetails.beneficiaryHouseholdSize}}
+          </ul>
+          <ul>
+            {{caseDetails.beneficiaryPaymentType}}
+          </ul>
+        </div>
         <br />
       </div>
       <div class="sectionHeadingWhite">NOTES</div>
```
Comment (farablanco) [inline src/pages/Details.vue:215]: We are missing the section on related cases

File: src/pages/Details.vue [hunk 8]
```diff
@@ -187,42 +244,148 @@ export default {
...
-    //       organisation: data.referee.organisation
-    //         ? data.referee.organisation
-    //         : '-',
-    //       lastUpdate: dayjs(data.updatedAt).format('DD/MM/YYYY'),
-    //     }
-    //   })
-    //   // Assign it to Vue data
-    //   table.fetchedItems = transformedData
-    // }
+    // const xx = props
```
Comment (farablanco) [inline src/pages/Details.vue]: remove these

File: src/pages/Details.vue [hunk 9]
```diff
@@ -187,42 +244,148 @@ export default {
...
+    // console.log(props) // { user: '' }
 
+    const refereeStatus = reactive([{//key: the array
```
Comment (farablanco) [inline src/pages/Details.vue]: is this needed?

File: src/pages/Details.vue [hunk 10]
```diff
@@ -187,42 +244,148 @@ export default {
...
+    }])
+    const amountGranted = reactive([{//key: the array
```
Comment (farablanco) [inline src/pages/Details.vue]: is this needed?

File: src/pages/Details.vue [hunk 11]
```diff
@@ -37,55 +42,67 @@
...
                 <select>
                   <option>Unverified</option>
-                  <option>Option 1</option>
-                  <option>Option 2</option>
-                  <option>Option 3</option>
+                  <option>Pending</option>
+                  <option>Verified</option>
                 </select>
               </div>
             </div>
```
Comment (farablanco) [inline src/pages/Details.vue:104]: Amount granted and reference status do  not reflect value from DB
Screenshot 2021-03-13 at 6 45 23 PM

**Review by farablanco — CHANGES_REQUESTED** [review]
Hi. I’ve completed the review. Almost there. I have a couple of comments. The most important ones would be on missing UI fields and UI fields not reflecting info from the DB. Please check comments and update.

File: src/components/CaseDetails.vue (hunk 1 above)
Comment (tongzxc) [reply src/components/CaseDetails.vue]: removed file

File: src/pages/Cases.vue (hunk 2 above)
Comment (tongzxc) [reply src/pages/Cases.vue]: removed console.log

File: src/pages/Details.vue (hunk 3 above)
Comment (tongzxc) [reply src/pages/Details.vue]: I don't have this in my file, should be removed already

File: src/pages/Details.vue (hunk 4 above)
Comment (tongzxc) [reply src/pages/Details.vue]: changed to Organisation

File: src/pages/Details.vue (hunk 5 above)
Comment (tongzxc) [reply src/pages/Details.vue]: changed to REFERENCE

File: src/pages/Details.vue (hunk 6 above)
Comment (tongzxc) [reply src/pages/Details.vue]: changed to Reference Status

File: src/pages/Details.vue (hunk 7 above)
Comment (tongzxc) [reply src/pages/Details.vue:215]: updated this section to include related cases

File: src/pages/Details.vue (hunk 8 above)
Comment (tongzxc) [reply src/pages/Details.vue]: removed

File: src/pages/Details.vue (hunk 9 above)
Comment (tongzxc) [reply src/pages/Details.vue]: removed

File: src/pages/Details.vue (hunk 10 above)
Comment (tongzxc) [reply src/pages/Details.vue]: removed

File: src/pages/Details.vue (hunk 11 above)
Comment (tongzxc) [reply src/pages/Details.vue:104]: have updated to show status and amount granted from db

**Review by farablanco — COMMENTED** [review]
added a refactor suggestion.
This can be picked up by whoever is going to work on the project next

**Review by farablanco — APPROVED** [review]
